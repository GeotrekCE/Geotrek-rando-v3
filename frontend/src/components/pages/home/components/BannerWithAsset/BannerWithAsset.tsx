import { FormattedMessage } from 'react-intl';
import styled from 'styled-components';
import { colorPalette } from 'stylesheet';
import { BannerCarousel } from '../BannerCarousel/BannerCarousel';

const WelcomeText = styled.span`
  text-shadow: 0 0 20px ${colorPalette.home.shadowOnImages};
`;

interface BannerSectionProps {
  videoUrl?: string;
  carouselUrls?: string[];
  pictureUrl?: string;
  shouldDisplayText: boolean;
}

interface BannerSectionWithoutAssetProps {
  shouldDisplayText: boolean;
  children: React.ReactNode;
}

const Banner: React.FC<BannerSectionWithoutAssetProps> = ({ shouldDisplayText, children }) => (
  <div id="banner" className="relative">
    {children}
    <div className="absolute bottom-0 w-full h-20 desktop:h-100 bg-gradient-to-t from-gradientOnImages to-transparent" />
    {shouldDisplayText && (
      <WelcomeText
        data-testid="text"
        className="text-white font-bold text-Mobile-H1 desktop:text-H1 text-center desktop:leading-tight
        absolute bottom-20 mx-10 desktop:bottom-40 desktop:items-end desktop:w-full desktop:mx-0"
      >
        <FormattedMessage id="home.welcome-text" />
      </WelcomeText>
    )}
  </div>
);

export const BannerWithAsset: React.FC<BannerSectionProps> = ({
  shouldDisplayText,
  videoUrl,
  carouselUrls,
  pictureUrl,
}) => {
  if (videoUrl !== undefined) {
    return (
      <Banner shouldDisplayText={shouldDisplayText}>
        <video
          id="banner-video"
          autoPlay
          muted
          loop
          src={videoUrl}
          className="object-cover object-center overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
          data-testid="video"
        />
      </Banner>
    );
  } else if (carouselUrls !== undefined) {
    return (
      <Banner shouldDisplayText={shouldDisplayText}>
        <BannerCarousel picturesUrl={carouselUrls} />
      </Banner>
    );
  } else if (pictureUrl !== undefined) {
    return (
      <Banner shouldDisplayText={shouldDisplayText}>
        <img
          id="banner-image"
          src={pictureUrl}
          className="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
          data-testid="image"
        />
      </Banner>
    );
  } else {
    return null;
  }
};
