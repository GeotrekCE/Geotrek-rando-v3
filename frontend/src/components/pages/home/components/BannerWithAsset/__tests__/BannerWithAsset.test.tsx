import { render } from 'services/testing/reactTestingLibraryWrapper';
import { BannerWithAsset } from '../BannerWithAsset';

// CF https://github.com/testing-library/react-testing-library/issues/470
Object.defineProperty(HTMLMediaElement.prototype, 'muted', {
  set: () => true,
});

describe('BannerWithAsset', () => {
  it('should well render with a video', () => {
    const component = render(<BannerWithAsset shouldDisplayText videoUrl="test.mp4" />);
    const asset = component.queryByTestId('video');

    expect(asset).toBeInTheDocument();

    expect(component).toMatchSnapshot();
  });

  it('should well render with a carousel', () => {
    const component = render(
      <BannerWithAsset shouldDisplayText carouselUrls={['test1.jpg', 'test2.jpg']} />,
    );
    const asset = component.queryByTestId('carousel');

    expect(asset).toBeInTheDocument();

    expect(component).toMatchSnapshot();
  });

  it('should well render with a single picture', () => {
    const component = render(<BannerWithAsset shouldDisplayText pictureUrl="test0.jpg" />);
    const asset = component.queryByTestId('image');

    expect(asset).toBeInTheDocument();

    expect(component).toMatchSnapshot();
  });

  test.each([true, false])('should adapt text if shouldDisplayText is %p', actual => {
    const component = render(<BannerWithAsset shouldDisplayText={actual} pictureUrl="image.jpg" />);

    const foundText = component.queryByTestId('text');

    expect(!!foundText).toBe(actual);
  });
});
