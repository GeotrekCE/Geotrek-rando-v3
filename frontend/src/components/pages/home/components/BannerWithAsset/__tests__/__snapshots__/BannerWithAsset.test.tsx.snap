// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerWithAsset should well render with a carousel 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative"
        id="banner"
      >
        <div
          data-testid="carousel"
          id="banner_carousel"
        >
          <div
            class="slick-slider h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full slick-initialized"
            dir="ltr"
          >
            <div
              class="Carousel__StyledArrow-sc-1tck7ng-1 Carousel__LargeStyledArrow-sc-1tck7ng-3 BannerCarousel__BannerStyledArrow-sc-q9y4d7-0 BannerCarousel__StyledLeftArrow-sc-q9y4d7-2 hBtEpW fuiwDG nGhjU MQxHa slick-arrow slick-prev"
            />
            <div
              class="slick-list"
            >
              <div
                class="slick-track"
                style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
              >
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="-1"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt=""
                      class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                      src="test2.jpg"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="false"
                  class="slick-slide slick-active slick-current"
                  data-index="0"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt=""
                      class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                      src="test1.jpg"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide"
                  data-index="1"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt=""
                      class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                      src="test2.jpg"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="2"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt=""
                      class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                      src="test1.jpg"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="3"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt=""
                      class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                      src="test2.jpg"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="Carousel__StyledArrow-sc-1tck7ng-1 Carousel__LargeStyledArrow-sc-1tck7ng-3 BannerCarousel__BannerStyledArrow-sc-q9y4d7-0 BannerCarousel__StyledRightArrow-sc-q9y4d7-1 hBtEpW fuiwDG nGhjU gSQeFB slick-arrow slick-next"
            />
          </div>
        </div>
        <div
          class="absolute bottom-0 w-full h-20 desktop:h-100 bg-gradient-to-t from-gradientOnImages to-transparent"
        />
        <h1
          class="BannerWithAsset__WelcomeText-sc-189knhl-0 dVXbNq text-white font-bold text-Mobile-H1 desktop:text-H1 text-center desktop:leading-tight absolute bottom-20 desktop:bottom-40 px-10 desktop:px-10vw w-full"
          data-testid="text"
        >
          L’écoguide numérique et local qui présente mes découvertes du territoire !
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative"
      id="banner"
    >
      <div
        data-testid="carousel"
        id="banner_carousel"
      >
        <div
          class="slick-slider h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full slick-initialized"
          dir="ltr"
        >
          <div
            class="Carousel__StyledArrow-sc-1tck7ng-1 Carousel__LargeStyledArrow-sc-1tck7ng-3 BannerCarousel__BannerStyledArrow-sc-q9y4d7-0 BannerCarousel__StyledLeftArrow-sc-q9y4d7-2 hBtEpW fuiwDG nGhjU MQxHa slick-arrow slick-prev"
          />
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="true"
                class="slick-slide slick-cloned"
                data-index="-1"
                style="width: 0px;"
                tabindex="-1"
              >
                <div>
                  <img
                    alt=""
                    class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                    src="test2.jpg"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                  />
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <img
                    alt=""
                    class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                    src="test1.jpg"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                  />
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="1"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <img
                    alt=""
                    class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                    src="test2.jpg"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                  />
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide slick-cloned"
                data-index="2"
                style="width: 0px;"
                tabindex="-1"
              >
                <div>
                  <img
                    alt=""
                    class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                    src="test1.jpg"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                  />
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide slick-cloned"
                data-index="3"
                style="width: 0px;"
                tabindex="-1"
              >
                <div>
                  <img
                    alt=""
                    class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
                    src="test2.jpg"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="Carousel__StyledArrow-sc-1tck7ng-1 Carousel__LargeStyledArrow-sc-1tck7ng-3 BannerCarousel__BannerStyledArrow-sc-q9y4d7-0 BannerCarousel__StyledRightArrow-sc-q9y4d7-1 hBtEpW fuiwDG nGhjU gSQeFB slick-arrow slick-next"
          />
        </div>
      </div>
      <div
        class="absolute bottom-0 w-full h-20 desktop:h-100 bg-gradient-to-t from-gradientOnImages to-transparent"
      />
      <h1
        class="BannerWithAsset__WelcomeText-sc-189knhl-0 dVXbNq text-white font-bold text-Mobile-H1 desktop:text-H1 text-center desktop:leading-tight absolute bottom-20 desktop:bottom-40 px-10 desktop:px-10vw w-full"
        data-testid="text"
      >
        L’écoguide numérique et local qui présente mes découvertes du territoire !
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BannerWithAsset should well render with a single picture 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative"
        id="banner"
      >
        <img
          alt=""
          class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
          data-testid="image"
          id="banner-image"
          src="test0.jpg"
        />
        <div
          class="absolute bottom-0 w-full h-20 desktop:h-100 bg-gradient-to-t from-gradientOnImages to-transparent"
        />
        <h1
          class="BannerWithAsset__WelcomeText-sc-189knhl-0 dVXbNq text-white font-bold text-Mobile-H1 desktop:text-H1 text-center desktop:leading-tight absolute bottom-20 desktop:bottom-40 px-10 desktop:px-10vw w-full"
          data-testid="text"
        >
          L’écoguide numérique et local qui présente mes découvertes du territoire !
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative"
      id="banner"
    >
      <img
        alt=""
        class="object-cover object-top overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
        data-testid="image"
        id="banner-image"
        src="test0.jpg"
      />
      <div
        class="absolute bottom-0 w-full h-20 desktop:h-100 bg-gradient-to-t from-gradientOnImages to-transparent"
      />
      <h1
        class="BannerWithAsset__WelcomeText-sc-189knhl-0 dVXbNq text-white font-bold text-Mobile-H1 desktop:text-H1 text-center desktop:leading-tight absolute bottom-20 desktop:bottom-40 px-10 desktop:px-10vw w-full"
        data-testid="text"
      >
        L’écoguide numérique et local qui présente mes découvertes du territoire !
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BannerWithAsset should well render with a video 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative"
        id="banner"
      >
        <video
          autoplay=""
          class="object-cover object-center overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
          data-testid="video"
          id="banner-video"
          loop=""
          src="test.mp4"
        />
        <div
          class="absolute bottom-0 w-full h-20 desktop:h-100 bg-gradient-to-t from-gradientOnImages to-transparent"
        />
        <h1
          class="BannerWithAsset__WelcomeText-sc-189knhl-0 dVXbNq text-white font-bold text-Mobile-H1 desktop:text-H1 text-center desktop:leading-tight absolute bottom-20 desktop:bottom-40 px-10 desktop:px-10vw w-full"
          data-testid="text"
        >
          L’écoguide numérique et local qui présente mes découvertes du territoire !
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative"
      id="banner"
    >
      <video
        autoplay=""
        class="object-cover object-center overflow-hidden h-bannerSectionMobile desktop:h-bannerSectionDesktop w-full"
        data-testid="video"
        id="banner-video"
        loop=""
        src="test.mp4"
      />
      <div
        class="absolute bottom-0 w-full h-20 desktop:h-100 bg-gradient-to-t from-gradientOnImages to-transparent"
      />
      <h1
        class="BannerWithAsset__WelcomeText-sc-189knhl-0 dVXbNq text-white font-bold text-Mobile-H1 desktop:text-H1 text-center desktop:leading-tight absolute bottom-20 desktop:bottom-40 px-10 desktop:px-10vw w-full"
        data-testid="text"
      >
        L’écoguide numérique et local qui présente mes découvertes du territoire !
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
